### ISSUE TITLE
Input is not recognized for some buttons

### ISSUE THREAD

**Issue by zach-morris** [kodi-game/game.libretro.genplus#5]
Version tested: RetroPlayer 18 alpha1 2018-02-11

Issue:  It looks like genplus is linked to msx controller.  Up/Down/Left/Right for example do not work and register as game.controller.msx.joystick.  This is true if genplus is playing either Genesis Games or SG-1000 games

```
21:27:33.721 T:140735305183232    INFO: GAME: ---------------------------------------
21:27:33.721 T:140735305183232    INFO: GAME: Base Width:   256
21:27:33.721 T:140735305183232    INFO: GAME: Base Height:  192
21:27:33.722 T:140735305183232    INFO: GAME: Max Width:    720
21:27:33.722 T:140735305183232    INFO: GAME: Max Height:   576
21:27:33.722 T:140735305183232    INFO: GAME: Aspect Ratio: 1.523810
21:27:33.722 T:140735305183232    INFO: GAME: FPS:          59.922743
21:27:33.722 T:140735305183232    INFO: GAME: Sample Rate:  44100.000000
21:27:33.722 T:140735305183232    INFO: GAME: Region:       NTSC
21:27:33.722 T:140735305183232    INFO: GAME: ---------------------------------------
...

21:27:35.372 T:123145315336192   DEBUG: BUTTON [ 6 ] on "PS(R) Gamepad" pressed
21:27:35.372 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.default pressed (ignored)
21:27:35.372 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.msx.joystick pressed (handled)
21:27:35.563 T:123145315336192   DEBUG: BUTTON [ 6 ] on "PS(R) Gamepad" released
21:27:35.563 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.default released
21:27:35.564 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.msx.joystick released
21:27:36.277 T:123145315336192   DEBUG: BUTTON [ 3 ] on "PS(R) Gamepad" pressed
21:27:36.278 T:123145315336192   DEBUG: FEATURE [ start ] on game.controller.default pressed (ignored)
21:27:36.278 T:123145315336192   DEBUG: FEATURE [ start ] on game.controller.default pressed (handled)
21:27:36.513 T:123145315336192   DEBUG: BUTTON [ 3 ] on "PS(R) Gamepad" released
21:27:36.514 T:123145315336192   DEBUG: FEATURE [ start ] on game.controller.default released
21:27:37.670 T:123145315336192   DEBUG: Previous line repeats 1 times.
21:27:37.671 T:123145315336192   DEBUG: BUTTON [ 13 ] on "PS(R) Gamepad" pressed
21:27:37.671 T:123145315336192   DEBUG: FEATURE [ b ] on game.controller.default pressed (ignored)
21:27:37.671 T:123145315336192   DEBUG: FEATURE [ b ] on game.controller.msx.joystick pressed (handled)
21:27:37.770 T:123145321238528   DEBUG: CAddonSettings[service.nextup.notification]: loading setting definitions
21:27:37.770 T:123145321238528   DEBUG: CAddonSettings[service.nextup.notification]: trying to load setting definitions from old format...
21:27:37.829 T:123145315336192   DEBUG: BUTTON [ 13 ] on "PS(R) Gamepad" released
21:27:37.829 T:123145315336192   DEBUG: FEATURE [ b ] on game.controller.default released
21:27:37.829 T:123145315336192   DEBUG: FEATURE [ b ] on game.controller.msx.joystick released
21:27:38.083 T:123145315336192   DEBUG: BUTTON [ 14 ] on "PS(R) Gamepad" pressed
21:27:38.083 T:123145315336192   DEBUG: FEATURE [ a ] on game.controller.default pressed (ignored)
21:27:38.083 T:123145315336192   DEBUG: FEATURE [ a ] on game.controller.msx.joystick pressed (handled)
21:27:38.206 T:123145315336192   DEBUG: BUTTON [ 14 ] on "PS(R) Gamepad" released
21:27:38.207 T:123145315336192   DEBUG: FEATURE [ a ] on game.controller.default released
21:27:38.207 T:123145315336192   DEBUG: FEATURE [ a ] on game.controller.msx.joystick released
21:27:38.414 T:140735305183232   DEBUG: ------ Window Deinit (Pointer.xml) ------
21:27:42.762 T:123145315336192   DEBUG: BUTTON [ 6 ] on "PS(R) Gamepad" pressed
21:27:42.762 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.default pressed (ignored)
21:27:42.762 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.msx.joystick pressed (handled)
21:27:42.774 T:123145321238528   DEBUG: CAddonSettings[service.nextup.notification]: loading setting definitions
21:27:42.774 T:123145321238528   DEBUG: CAddonSettings[service.nextup.notification]: trying to load setting definitions from old format...
21:27:42.918 T:123145315336192   DEBUG: BUTTON [ 6 ] on "PS(R) Gamepad" released
21:27:42.918 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.default released
21:27:42.918 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.msx.joystick released
21:27:43.010 T:123145315336192   DEBUG: BUTTON [ 6 ] on "PS(R) Gamepad" pressed
21:27:43.010 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.default pressed (ignored)
21:27:43.010 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.msx.joystick pressed (handled)
21:27:43.165 T:123145315336192   DEBUG: BUTTON [ 6 ] on "PS(R) Gamepad" released
21:27:43.166 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.default released
21:27:43.167 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.msx.joystick released
21:27:43.284 T:123145315336192   DEBUG: BUTTON [ 6 ] on "PS(R) Gamepad" pressed
21:27:43.284 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.default pressed (ignored)
21:27:43.284 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.msx.joystick pressed (handled)
21:27:43.393 T:123145315336192   DEBUG: BUTTON [ 6 ] on "PS(R) Gamepad" released
21:27:43.394 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.default released
21:27:43.394 T:123145315336192   DEBUG: FEATURE [ down ] on game.controller.msx.joystick released
21:27:43.650 T:123145315336192   DEBUG: BUTTON [ 5 ] on "PS(R) Gamepad" pressed
21:27:43.650 T:123145315336192   DEBUG: FEATURE [ right ] on game.controller.default pressed (ignored)
21:27:43.651 T:123145315336192   DEBUG: FEATURE [ right ] on game.controller.msx.joystick pressed (handled)
21:27:43.720 T:123145315336192   DEBUG: BUTTON [ 5 ] on "PS(R) Gamepad" released
21:27:43.720 T:123145315336192   DEBUG: FEATURE [ right ] on game.controller.default released
21:27:43.720 T:123145315336192   DEBUG: FEATURE [ right ] on game.controller.msx.joystick released
21:27:43.728 T:123145309356032   DEBUG: Saving savestate to /Users/xxx/Library/Application Support/Kodi/userdata/addon_data/plugin.program.iarl/temp_iarl/Bomberman Special (Taiwan).sav
21:27:43.729 T:123145309356032   DEBUG: Saved state to /Users/xxx/Library/Application Support/Kodi/userdata/addon_data/plugin.program.iarl/temp_iarl/Bomberman Special (Taiwan).sav
21:27:43.859 T:123145315336192   DEBUG: BUTTON [ 5 ] on "PS(R) Gamepad" pressed
21:27:43.859 T:123145315336192   DEBUG: FEATURE [ right ] on game.controller.default pressed (ignored)
21:27:43.859 T:123145315336192   DEBUG: FEATURE [ right ] on game.controller.msx.joystick pressed (handled)
21:27:43.946 T:123145315336192   DEBUG: BUTTON [ 5 ] on "PS(R) Gamepad" released
21:27:43.946 T:123145315336192   DEBUG: FEATURE [ right ] on game.controller.default released
21:27:43.946 T:123145315336192   DEBUG: FEATURE [ right ] on game.controller.msx.joystick released
21:27:44.050 T:123145315336192   DEBUG: BUTTON [ 5 ] on "PS(R) Gamepad" pressed
21:27:44.051 T:123145315336192   DEBUG: FEATURE [ right ] on game.controller.default pressed (ignored)
21:27:44.051 T:123145315336192   DEBUG: FEATURE [ right ] on game.controller.msx.joystick pressed (handled)
21:27:44.158 T:123145315336192   DEBUG: BUTTON [ 5 ] on "PS(R) Gamepad" released
21:27:44.158 T:123145315336192   DEBUG: FEATURE [ right ] on game.controller.default released
21:27:44.159 T:123145315336192   DEBUG: FEATURE [ right ] on game.controller.msx.joystick released
21:27:44.333 T:123145315336192   DEBUG: BUTTON [ 4 ] on "PS(R) Gamepad" pressed
21:27:44.333 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.default pressed (ignored)
21:27:44.333 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.msx.joystick pressed (handled)
21:27:44.451 T:123145315336192   DEBUG: BUTTON [ 4 ] on "PS(R) Gamepad" released
21:27:44.451 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.default released
21:27:44.451 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.msx.joystick released
21:27:44.556 T:123145315336192   DEBUG: BUTTON [ 4 ] on "PS(R) Gamepad" pressed
21:27:44.557 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.default pressed (ignored)
21:27:44.557 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.msx.joystick pressed (handled)
21:27:44.608 T:123145315336192   DEBUG: BUTTON [ 4 ] on "PS(R) Gamepad" released
21:27:44.608 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.default released
21:27:44.608 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.msx.joystick released
21:27:44.747 T:123145315336192   DEBUG: BUTTON [ 4 ] on "PS(R) Gamepad" pressed
21:27:44.747 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.default pressed (ignored)
21:27:44.747 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.msx.joystick pressed (handled)
21:27:44.817 T:123145315336192   DEBUG: BUTTON [ 4 ] on "PS(R) Gamepad" released
21:27:44.818 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.default released
21:27:44.818 T:123145315336192   DEBUG: FEATURE [ up ] on game.controller.msx.joystick released
21:27:45.026 T:123145315336192   DEBUG: BUTTON [ 14 ] on "PS(R) Gamepad" pressed
21:27:45.026 T:123145315336192   DEBUG: FEATURE [ a ] on game.controller.default pressed (ignored)
21:27:45.026 T:123145315336192   DEBUG: FEATURE [ a ] on game.controller.msx.joystick pressed (handled)
21:27:45.182 T:123145315336192   DEBUG: BUTTON [ 14 ] on "PS(R) Gamepad" released
21:27:45.182 T:123145315336192   DEBUG: FEATURE [ a ] on game.controller.default released
21:27:45.183 T:123145315336192   DEBUG: FEATURE [ a ] on game.controller.msx.joystick released
21:27:45.478 T:123145315336192   DEBUG: BUTTON [ 13 ] on "PS(R) Gamepad" pressed
21:27:45.479 T:123145315336192   DEBUG: FEATURE [ b ] on game.controller.default pressed (ignored)
21:27:45.479 T:123145315336192   DEBUG: FEATURE [ b ] on game.controller.msx.joystick pressed (handled)
21:27:45.600 T:123145315336192   DEBUG: BUTTON [ 13 ] on "PS(R) Gamepad" released
21:27:45.601 T:123145315336192   DEBUG: FEATURE [ b ] on game.controller.default released
21:27:45.602 T:123145315336192   DEBUG: FEATURE [ b ] on game.controller.msx.joystick released
```

**Comment by garbear** [comment]
Can you try the 2018-02-15 build?

**Comment by zach-morris** [comment]
Confirmed fixed in 18.0-ALPHA1 Git:20180215-276e1a4
